instrumentation/kamon-redis/src/main/scala/kamon/instrumentation/jedis/JedisInstrumentation.scala

```diff
@@ -14,36 +14,125 @@
  * limitations under the License.
  */
 
-package kamon.instrumentation.jedis
+package kamon
+package instrumentation
+package jedis
 
-import kamon.Kamon
+import kamon.context.Storage.Scope
+import kamon.tag.Lookups.plain
 import kamon.trace.Span
 import kanela.agent.api.instrumentation.InstrumentationBuilder
 import kanela.agent.libs.net.bytebuddy.asm.Advice
 import kanela.agent.libs.net.bytebuddy.description.method.MethodDescription
-import kanela.agent.libs.net.bytebuddy.matcher.ElementMatchers.isPublic
+import kanela.agent.libs.net.bytebuddy.matcher.ElementMatchers
+import kanela.agent.libs.net.bytebuddy.matcher.ElementMatchers.{isMethod, isPublic, isStatic, namedOneOf, not, takesArgument, takesArguments}
 
 class JedisInstrumentation extends InstrumentationBuilder {
+  onTypes("redis.clients.jedis.Jedis", "redis.clients.jedis.BinaryJedis")
+    .advise(
+      isMethod[MethodDescription]()
+        .and(isPublic[MethodDescription])
+        .and(not(isStatic[MethodDescription]))
+        .and(not(namedOneOf[MethodDescription](
+          "setDataSource",
+          "getDB",
+          "isConnected",
+          "connect",
+          "disconnect",
+          "resetState",
+          "getClient",
+          "getConnection",
+          "isConnected",
+          "isBroken",
+          "close",
+          "toString",
+          "hashCode"
+        ))),
+      classOf[ClientOperationsAdvice])
+
+
+  onType("redis.clients.jedis.Protocol")
+    .advise(
+      method("sendCommand")
+        .and(isPublic[MethodDescription])
+        .and(isStatic[MethodDescription])
+        .and(takesArguments(3)),
+      classOf[SendCommandAdvice])
+
   onType("redis.clients.jedis.Protocol")
-    .advise(method("sendCommand").and(isPublic[MethodDescription]), classOf[SendCommandAdvice])
+    .when(classIsPresent("redis.clients.jedis.CommandArguments"))
+    .advise(
+      method("sendCommand")
+        .and(isPublic[MethodDescription])
+        .and(isStatic[MethodDescription])
+        .and(takesArguments(2)),
+      classOf[SendCommandAdviceForJedis4])
```

Comment:
I guess this is for supporting different Jedis versions, but a few comments explaining, will help the next developer who looks at this :)

Reply:
Pushed a change for it. Thanks for the hint Joao!